src/containers/Settings/UserSettings.js

```diff
@@ -0,0 +1,284 @@
+              <div className="col-xs-6">
+                <Field
+                  disabled={savingSettings}
+                  name="website"
+                  component={FormTextInput}
+                  label="Personal website"
+                  placeholder="https://example.com"
+                  validate={[validators.maxLength(128)]}
+                />
+              </div>
+              <div className="col-xs-6">
+                <Field
+                  disabled={savingSettings}
+                  name="twitter"
+                  component={FormTextInput}
+                  label="Twitter"
+                  placeholder="@ethBounties"
+                  validate={[validators.maxLength(128)]}
+                />
+              </div>
+            </div>
+          </FormSection.InputGroup>
+          <FormSection.InputGroup>
+            <div className="row">
+              <div className="col-xs-6">
+                <Field
+                  disabled={savingSettings}
+                  name="github"
+                  component={FormTextInput}
+                  label="Github"
+                  placeholder="@vbuterin"
+                  validate={[validators.maxLength(128)]}
+                />
+              </div>
+              <div className="col-xs-6">
+                <Field
+                  disabled={savingSettings}
+                  name="linkedin"
+                  component={FormTextInput}
+                  label="LinkedIn"
+                  placeholder="https://linkedin.com/in/vbuterin"
+                  validate={[validators.maxLength(128)]}
+                />
+              </div>
+            </div>
+          </FormSection.InputGroup>
+        </FormSection.Section>
+      </FormSection>
+      <PageCard.Break />
+      <div className={styles.buttons}>
+        <Button
+          type="primary"
+          disabled={uploading || (submitFailed && invalid)}
+          loading={savingSettings}
+        >
+          Update Profile
+        </Button>
+        {submitFailed && invalid ? (
+          <Text inputLabel color="red" className={styles.submitError}>
+            Fix errors before submitting.
+          </Text>
+        ) : null}
+      </div>
+    </form>
+  );
+};
+
+const mapStateToProps = state => {
+  const rootUpload = rootUploadSelector(state);
+  const uploadState = rootUpload[UPLOAD_KEY] || {};
+  const currentUser = getCurrentUserSelector(state);
+
+  return {
+    initialValues: {
+      ...currentUser,
+      languages: currentUser.languages.join(', ')
+    },
+    uploading: uploadState.uploading || false,
+    uploaded: uploadState.uploaded || false,
+    ipfsHash: uploadState.uploaded ? uploadState.ipfsHash : null,
+    fileName: uploadState.uploaded ? uploadState.fileName : null,
+    skills: skillsSelector(state),
+    savingSettings: settingsSelector(state).saving,
+    errorSavingSettings: settingsSelector(state).error,
+    profilePhotoURL: uploadState.uploaded
```

Comment:
we should just update the uploadState to have these fields... that way we can simplify this logic.